lib/routes/recovery-keys.js

```diff
@@ -0,0 +1,223 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict'
+
+const errors = require('../error')
+const isA = require('joi')
+const random = require('../crypto/random')
+const requestHelper = require('../routes/utils/request_helper')
+const validators = require('./validators')
+const HEX_STRING = validators.HEX_STRING
+
+module.exports = (log, db, Password, verifierVersion, customs) => {
+  return [
+    {
+      method: 'POST',
+      path: '/recoveryKeys',
+      config: {
+        auth: {
+          strategy: 'sessionToken'
+        },
+        validate: {
+          payload: {
+            recoveryKeyId: isA.string().max(32).required(),
+            recoveryData: isA.string().max(1024).required()
+          }
+        }
+      },
+      handler(request, reply) {
+        log.begin('createRecoveryKey', request)
+
+        const uid = request.auth.credentials.uid
+        const sessionToken = request.auth.credentials
+        const {recoveryKeyId, recoveryData} = request.payload
+
+        customs.check(request, 'createRecoveryKey')
+          .then(createRecoveryKey)
+          .then(emitMetrics)
+          .then(() => reply({}), reply)
+
+        function createRecoveryKey() {
+          if (sessionToken.tokenVerificationId) {
+            throw errors.unverifiedSession()
+          }
+
+          return db.createRecoveryKey(uid, recoveryKeyId, recoveryData)
+        }
+
+        function emitMetrics() {
+          log.info({
+            op: 'account.recoveryKey.created',
+            uid
+          })
+
+          return request.emitMetricsEvent('recoveryKey.created', {uid})
+        }
+      }
+    },
+    {
+      method: 'GET',
+      path: '/recoveryKeys/{recoveryKeyId}',
+      config: {
+        auth: {
+          strategy: 'accountResetToken'
+        },
+        validate: {
+          params: {
+            recoveryKeyId: isA.string().max(32).required()
+          }
+        }
+      },
+      handler(request, reply) {
+        log.begin('getRecoveryKey', request)
+
+        const uid = request.auth.credentials.uid
+        const recoveryKeyId = request.params.recoveryKeyId
+        let recoveryData
+
+        customs.check(request, 'getRecoveryKey')
```

Comment:
I'm confused by this and concerned that it seems to work without error...the signature for `customs.check` should be `customs.check(request, email, action)` so I'd expect this to fail closed due to the missing parameter.  Am I missing some magic that makes it work?